cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -1006,6 +1008,32 @@ public void grantKeyVaultAccessPolicyToServicePrincipal(String resourceGroupName
         });
     }
 
+    public boolean checkKeyVaultAccessPolicyForServicePrincipal(String resourceGroupName, String vaultName, String principalObjectId) {
+        return handleAuthException(() -> {
+            List<AccessPolicy> accessPolicies = azure.vaults()
+                    .getByResourceGroup(resourceGroupName, vaultName)
+                    .accessPolicies();
+            if (accessPolicies != null) {
+                for (int i = accessPolicies.size() - 1; i >= 0; i--) {
+                    AccessPolicyEntry accessPolicyEntry = accessPolicies.get(i).inner();
+                    if (checkKeyVaultAccessPolicyListForServicePrincipal(accessPolicyEntry, principalObjectId)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
```

Comment:
As mentioned earlier, I suggest extracting this whole code block as a package-private method annotated with `@VisibleForTesting` and then have it tested in the UT.